modules/op-configuring-pipelines-as-code-for-a-github-app.adoc

```diff
@@ -0,0 +1,79 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-pipelines-as-code.adoc
+
+:_content-type: PROCEDURE
+[id="configuring-pipelines-as-code-for-a-github-app_{context}"]
+= Configuring Pipelines as Code for a GitHub App 
+
+[role="_abstract"]
+GitHub Apps act as a point of integration with {pipelines-title}, and brings the advantage of Git based workflows to Tekton Pipelines. Cluster administrators can configure a single GitHub application for all users of the cluster. For GitHub Apps to work with Pipelines as Code, ensure that the webhook of the GitHub App to points to your Pipelines as Code controller route (or ingress endpoint) which listens for GitHub events.
+
+[id="configuring-github-app-for-pac"]
+== Configuring a GitHub App
+
+[discrete]
+.Procedure
+
+. Sign in to your GitHub account, go to **Settings** --> **Developer settings** --> **GitHub Apps**, and click **New GitHub App**.
+
+. Provide the following information in the GitHub App form: 
+
+** **GitHub Application Name**: `OpenShift Pipelines`
+** **Homepage URL**: OpenShift Console URL 
+** **Webhook URL**: The Pipelines as Code route or ingress URL. You can find it by executing the command `echo https://$(oc get route -n pipelines-as-code el-pipelines-as-code-interceptor -o jsonpath='{.spec.host}')`.
+** **Webhook secret**: An arbitrary secret. You can generate a secret by executing the command `openssl rand -hex 20`.
+
+. Select the following **Repository permissions**:
+
+** **Checks**: `Read & Write`
+** **Contents**: `Read & Write`
+** **Issues**: `Read & Write`
+** **Metadata**: `Readonly`
+** **Pull request**: `Read & Write`
+
+. Select the following **Organization permissions**:
+
+** **Members**: `Readonly`
+** **Plan**: `Readonly`
+
+. Select the following **User permissions**:
+
+** **Commit comment**
+** **Issue comment**
+** **Pull request**
+** **Pull request review**
+** **Pull request review comment**
+** **Push**
+
+. Click **Create GitHub App**.
+
+. On the **Details** page of the newly created GitHub App, note of the **App ID** displayed at the top.
+
+. In the **Private keys** section, click **Generate Private key** to automatically generate and download a private key for the GitHub app. Securely store the private key for future reference and usage.
+
+
+[id="configuring-pac-for-github-app"]
+== Configuring Pipelines as Code to access a GitHub App
+
```

Comment:
To follow mod docs guidelines, the following procedure should be removed from this file and put in is own PROCEDURE module file.

Reply:
It will be too much fragmentation and away from the context. I made the necessary workarounds to pass the guidelines.